mne/io/base.py

```diff
@@ -2009,3 +2009,112 @@ def _check_update_montage(info, montage):
+    system-specific read_raw_* function.
+
+
+    Parameters
+    ----------
+    input_fname : str | list of str
+        The path to raw file or a list of the raw files to treat as a Raw
+        instance. For files that have automatically been split, only the
+        name of the first file has to be specified. List of raw files is
+        currently only implemented for '.fif' files.
+    montage : None | str | instance of Montage | list of Montage instances
+        If None and eeg only, sensor locations are (0,0,0).
+        If str, this correspond to the path of a Montage instance. If instance,
+        it can be a Montage or a DigMontage. If a list, it can contain a list
+        of distinct montages: a Montage and DigMontage, not two Montages, not
+        two DigMontages.
+    eog : list or tuple of str
+        Names of channels or list of indices that should be designated
+        EOG channels. Values should correspond to channels listed within
+        the raw file.
+    misc : list or tuple of str
+        Names of channels or list of indices that should be designated
+        MISC channels. Values should correspond to channels listed within
+        the raw file.
+    reference : bool | str
+        If True, add average EEG reference projector (if it's not already
+        present).
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    raw : Instance of Raw
+        A Raw object.
+    """
+
+    from mne.io import (read_raw_brainvision, read_raw_bti, read_raw_edf,
+                        read_raw_egi, read_raw_fif, read_raw_kit)
+
+    if isinstance(input_fname, str):
+        fname, ext = os.path.splitext(input_fname)
+    elif isinstance(input_fname, list):
+        for f in input_fname:
+            fname, ext = os.path.splitext(input_fname)
+            err = "File must be '.fif', not %s." % ext
+            assert_equal(ext, '.fif', err)
+
+        if eog is not None:
+            eog_map = dict()
+            for s in eog:
+                eog_map[s] = 'eog'
+        if misc is not None:
+            misc_map = dict()
+            for s in misc:
+                misc_map[s] = 'misc'
+
+    # (MEG) KIT systems
+    if ext in ['.con', '.sqd']:
+        raw = read_raw_kit(input_fname, mrk=montage, eog=eog, misc=misc,
+                           preload=preload, verbose=verbose)
+        if eog is not None:
+            rename_sensor(raw.info, eog_map)
+        if misc is not None:
+            rename_sensor(raw.info, misc_map)
```

Comment:
I love it that we will have a common interface to read all file types.
However, I don't like the idea of the common `read_raw` function doing anything else but pass on the parameters to the respective `read_*` functions. Wouldn't it be better in this case to modify the `read_raw_kit` function to take an `eog` and `misc` parameter?